From 79579023e25ffcf4302f30d07d853d328f937680 Mon Sep 17 00:00:00 2001
From: Daniel Baluta <daniel.baluta@nxp.com>
Date: Wed, 18 Oct 2017 14:15:37 +0300
Subject: [PATCH 2667/5242] MLK-16607: arm64: fsl-imx8qm-mek: Add headphone
 detect property

commit  c9ae8b845645736e6a96e444c406662c1c7ed670 from
https://source.codeaurora.org/external/imx/linux-imx.git

With this patch 'Playback Volume' control is now usable and
we can notice that the sound volume changes.

 BuildInfo:
  - SCFW f5910b7d, IMX-MKIMAGE fb52c576, ATF a438801
  - U-Boot 2017.03-00047-g8fe8d6d

Reviewed-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
index 5adff2b..4bd5913 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
@@ -65,6 +65,13 @@
 		audio-codec = <&wm8960>;
 		asrc-controller = <&asrc0>;
 		codec-master;
+		/*
+		 * hp-det = <hp-det-pin hp-det-polarity>;
+		 * hp-det-pin: JD1 JD2  or JD3
+		 * hp-det-polarity = 0: hp detect high for headphone
+		 * hp-det-polarity = 1: hp detect high for speaker
+		 */
+		hp-det = <2 0>;
 		audio-routing =
 			"Headphone Jack", "HP_L",
 			"Headphone Jack", "HP_R",
-- 
1.7.9.5

